lib/vehicle.rb

```diff
@@ -6,13 +6,25 @@ class Vehicle
               :make,
               :model,
               :engine
+              
+  attr_accessor :plate_type, :registration_date
 
   def initialize(vehicle_details)
-    @vin = vehicle_details[:vin]
-    @year = vehicle_details[:year]
+    if vehicle_details.keys.include?(:vin_1_10)
```

Comment:
same goes here -- we want our initialize especially to remain pretty simple, so extracting if statements/logic into helper methods is a great option for simplifying this (see Facility note for more examples)

Reply:
Right, I’ll keep that in mind for next time. I’m very used to long elsif chains/switch-case statements but the helper methods would look much cleaner.